compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -233,14 +233,15 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         mutability: Mutability,
     ) -> InterpResult<'tcx, Pointer<M::Provenance>> {
         let alloc = Allocation::from_bytes(bytes, align, mutability);
-        self.allocate_raw_ptr(alloc, kind)
+        self.add_allocation(alloc, kind)
     }
 
-    pub fn allocate_raw_ptr(
+    pub fn add_allocation(
```

Comment:
Why is this called "add"? It's not clear to me why the new name is better, can you explain?

Reply:
I'm not super happy with the new name... but the old name was perpetually confusing since it sounded like it was talking about raw pointers. But really, the "_ptr" suffix here is to distinguish these from methods working on places, so this is the "pointer" version of adding a "raw allocation".

I called it "add" because it "adds" the allocation to memory. But thinking about it, for HashMap we call this `insert`. So maybe `insert_allocation` would be better?

Reply:
After thinking about this for too long, I think insert is better. At least it evokes the right signature, add doesn't really suggest anything.

Reply:
All right, I renamed it.